jopa-owl2java/src/main/java/cz/cvut/kbss/jopa/owl2java/prefix/PrefixMap.java

```diff
@@ -0,0 +1,137 @@
+        assert ontology.getOntologyID().getOntologyIRI().isPresent();
+        final IRI ontologyIri = ontology.getOntologyID().getOntologyIRI().get();
+        ontology.axioms(AxiomType.ANNOTATION_ASSERTION)
+                .filter(ax -> ax.getProperty().equals(annProperty) && ax.getValue().isLiteral() && ax.getSubject()
+                                                                                                     .isIRI())
+                .forEach(ax -> result.put(ax.getSubject().asIRI().get().getIRIString(), ax.getValue().asLiteral().get()
+                                                                                          .getLiteral()));
+        final OWLDataProperty dataProperty = df.getOWLDataProperty(prefixProperty);
+        ontology.axioms(AxiomType.DATA_PROPERTY_ASSERTION)
+                .filter(ax -> ax.getProperty().equals(dataProperty) && ax.getSubject().isIndividual())
+                .forEach(ax -> result.put(ax.getSubject().asOWLNamedIndividual().toStringID(), ax.getObject()
+                                                                                                 .getLiteral()));
+        if (!result.containsKey(ontologyIri.getIRIString())) {
+            remotePrefixResolver.resolvePrefix(ontologyIri).ifPresent(p -> result.put(ontologyIri.getIRIString(), p));
+        }
+        return result;
+    }
+
+    private Map<String, String> resolvePrefixesFromPrefixMappingFile(String mappingFilePath) {
+        if (mappingFilePath == null) {
+            return Collections.emptyMap();
+        }
+        final File file = new File(mappingFilePath);
+        try {
+            LOG.debug("Loading prefix mapping from file '{}'.", file);
+            final List<String> lines = Files.readAllLines(file.toPath(), StandardCharsets.UTF_8);
+            final Map<String, String> prefixMap = new HashMap<>(lines.size());
+            lines.forEach(line -> {
+                final String[] mapping = line.split("=");
+                if (mapping.length != 2) {
+                    LOG.warn("Mapping line '{}' does not correspond to the expected pattern '$namespace=$prefix'. Skipping it.", line);
+                }
+                prefixMap.put(mapping[0], mapping[1]);
+            });
+            return prefixMap;
+        } catch (IOException e) {
+            LOG.error("Unable to read prefix mapping file.", e);
+            throw new OWL2JavaException("Unable to read prefix mapping file.", e);
+        }
+    }
+
+    /**
+     * Gets prefix for an ontology with the specified IRI.
+     *
+     * @param ontologyIri Ontology IRI
+     * @return Resolved prefix, if available
+     */
+    public Optional<String> getPrefix(IRI ontologyIri) {
+        Objects.requireNonNull(ontologyIri);
+        return Optional.ofNullable(prefixes.get(ontologyIri.getIRIString()));
+    }
+
+    private static Map<String, String> defaultPrefixes() {
```

Comment:
Should not this be a default mapping file instead?

Reply:
I prefer keeping them as a constant so that they can be overriden. Moreover, this way I can reference existing constants.